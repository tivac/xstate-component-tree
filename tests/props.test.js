import describe from "./util/describe.js";
import { getTree } from "./util/trees.js";
import component from "./util/component.js";
import { snapshot } from "./util/snapshot.js";

describe("props", (it) => {
    it("should support static props", async () => {
        const tree = await getTree({
            initial : "one",

            states : {
                one : {
                    meta : {
                        component : component("one"),
                        props     : {
                            fooga : 1,
                            booga : 2,
                        },
                    },

                    initial : "two",

                    states : {
                        two : {
                            meta : {
                                component : component("two"),
                                props     : {
                                    wooga : 1,
                                    tooga : 2,
                                },
                            },
                        },
                    },
                },
            },
        });

        snapshot(tree, `[
            [Object: null prototype] {
                path: "one",
                component: [Function: one],
                props: {
                    fooga: 1,
                    booga: 2
                },
                children: [
                    [Object: null prototype] {
                        path: "one.two",
                        component: [Function: two],
                        props: {
                            wooga: 1,
                            tooga: 2
                        },
                        children: []
                    }
                ]
            }
        ]`);
    });
});
