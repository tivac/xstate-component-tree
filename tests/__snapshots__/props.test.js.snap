// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`xstate-component-tree props should support static props 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "children": Array [],
        "component": "two",
        "path": "one.two",
        "props": Object {
          "tooga": 2,
          "wooga": 1,
        },
      },
    ],
    "component": "one",
    "path": "one",
    "props": Object {
      "booga": 2,
      "fooga": 1,
    },
  },
]
`;
